test/python/primitives/test_primitive.py

```diff
@@ -12,12 +12,16 @@
 
 """Tests for BasePrimitive."""
 
-from ddt import ddt, data, unpack
+import json
 
-from numpy import array, int32, int64, float32, float64
+from ddt import data, ddt, unpack
+from numpy import array, float32, float64, int32, int64
 
+from qiskit import QuantumCircuit, pulse, transpile
 from qiskit.circuit.random import random_circuit
 from qiskit.primitives.base.base_primitive import BasePrimitive
+from qiskit.primitives.utils import _circuit_key
```

Comment:
I'm not a big fan of these tests of internal features, because generally they make it harder for the internal implementation to evolve without changing the tests.  Ideally we should only be testing the outward public behaviour, and not the particulars of _how_ that's achieved.  If the tests are too tightly coupled, then any change to the internal implementation has to change the tests, and that makes it much easier for bugs and regressions to sneak in; we have to verify that the new tests are equivalent, and cover the same ground.

That said, this is a single function, and if you think it's much better to test this way, I won't block this PR on it.

I'd potentially attempt to rewrite the tests by using a custom subclass of the primitives for testing that has its `_run` (or whatever it's called) method defined to make assertions about the normalised circuits it gets back from the base class's handling.  That's testing public parts of the subclassing API, rather than this internal detail of _how_ that's done.

*edit*: I also see now that these tests are just moved from one file to another, rather than actually written new.  I'm still not a fan, but it does move this change out-of-scope of this PR.